ext.py

```diff
@@ -15,10 +22,41 @@ def ext_tags(value, rule_obj, path):
 
 # Check if provided resources are in resources.yaml
 def ext_resources(value, rule_obj, path):
-    resources = yaml.safe_load(open('resources.yaml'))
     if value not in resources:
         print('Invalid resource!', value)
         return False
 
     # If we're here, all resources were ok
     return True
+
+def ext_valid_bucket_regions(value, rule_obj, path):
+
+    # Make sure this is a dict, and a bucket, then validate the region
+    if isinstance(value, dict) and 'Type' in value and value['Type'] == 'S3 Bucket':
+        bucket = value['ARN']
+        parts = bucket.split(':::')
+        bucket = parts[1]
+        parts = bucket.split('/')
+        bucket = parts[0]
+        url = "https://{}.s3.amazonaws.com".format(bucket)
+
+        # Get the headers for this bucket.
+        # Verify=False because the wildcard matching doesn't work for buckets with '.'
+        r = requests.head(url, verify=False)
+        
+        if r.status_code == requests.codes.not_found:
+            print("Bucket {} doesn't exist or there was a momentary glitch".format(bucket))
```

Comment:
What's the logic in not failing here?

Reply:
Some deprecated buckets no longer exist. So we'd need to move the schema validation for bucket region up to the top level to be able to capture that info in the check.

Reply:
Would it make more sense to just skip the check if deprecated property in the YAML is true?